### PR TITLE
fix(emotion): fix text direction not working on simple html elements

### PR DESCRIPTION
**Pull request by Brailor**

### CODE REVIEW

**Review by HerrTopi — APPROVED** [review]

File: packages/emotion/src/InstUISettingsProvider/index.tsx [hunk 1]
```diff
@@ -27,14 +27,26 @@ import { TextDirectionContext } from '@instructure/ui-i18n'
...
+} from '@instructure/ui-react-utils'
 import type { DeterministicIdProviderValue } from '@instructure/ui-react-utils'
+import type { AsElementType } from '@instructure/shared-types'
 
 type InstUIProviderProps = {
   theme?: ThemeOrOverride
-  dir?: 'ltr' | 'rtl' // TODO allow "auto" too
   instanceCounterMap?: DeterministicIdProviderValue
-}
+} & (
```
Comment (ke1k0) [inline packages/emotion/src/InstUISettingsProvider/index.tsx:40]: For me TS doesn't underline if I add the `as` prop without the `dir` :(

File: packages/emotion/src/InstUISettingsProvider/index.tsx [hunk 2]
```diff
@@ -136,12 +150,25 @@ function InstUISettingsProvider({
       </ThemeProvider>
     </DeterministicIdContextProvider>
   )
+
+  if (dir) {
+    const Element = getElementType(InstUISettingsProvider, { as })
+    providers = <Element dir={finalDir}>{providers}</Element>
+  } else if (as && process.env.NODE_ENV !== 'production') {
+    console.warn(
+      "The 'as' property should be used in conjunction with the 'dir' property!"
```
Comment (ke1k0) [inline packages/emotion/src/InstUISettingsProvider/index.tsx:159]: It might be overkill but we could write a test for this warning too.

Comment (Brailor) [reply packages/emotion/src/InstUISettingsProvider/index.tsx:163]: You are correct, I added a test case for this.

File: packages/emotion/src/InstUISettingsProvider/index.tsx (hunk 1 above)
Comment (Brailor) [reply packages/emotion/src/InstUISettingsProvider/index.tsx:53]: Should be fixed:
Screenshot 2022-03-24 at 10 23 48

**Review by ke1k0 — APPROVED** [review]

**Pull request merged**